aci/resource_aci_dhcpoptionpol.go

```diff
@@ -296,50 +293,55 @@ func resourceAciDHCPOptionPolicyUpdate(ctx context.Context, d *schema.ResourceDa
 	}
 
 	if d.HasChange("dhcp_option") {
-		dhcpOption := d.Get("dhcp_option_ids").([]interface{})
-		for _, val := range dhcpOption {
-			dhcpOptionDN := val.(string)
-			err := aciClient.DeleteByDn(dhcpOptionDN, "dhcpOption")
-			if err != nil {
-				return diag.FromErr(err)
+		old_options, new_options := d.GetChange("dhcp_option")
+		getDifference := differenceInMaps(old_options.(*schema.Set), new_options.(*schema.Set))
+		listDNToDelete := make([]string, 0)
+		for i := 0; i < len(getDifference); i++ {
+			for key, value := range getDifference[i].(map[string]interface{}) {
+				if key == "id" {
+					listDNToDelete = append(listDNToDelete, value.(string))
+				}
 			}
 		}
 
-		options := d.Get("dhcp_option")
-		dhcpOptionIDS := make([]string, 0, 1)
-
-		dhcpOptions := options.([]interface{})
-		for _, val := range dhcpOptions {
-			dhcpOptionAttr := models.DHCPOptionAttributes{}
-			dhcpOption := val.(map[string]interface{})
-
-			name := dhcpOption["name"].(string)
+		for _, value := range listDNToDelete {
+			if value != "" {
+				err := aciClient.DeleteByDn(value, "dhcpOption")
+				if err != nil {
+					return diag.FromErr(err)
+				}
```

Comment:
Any reason to do the delete in another loop and not directly in the previous one?

Reply:
This section is not required if I use for loop with range above.